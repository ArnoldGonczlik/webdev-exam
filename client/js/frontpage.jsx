function Frontpage() {
    return <div>Change test!!</div>
}

export default Frontpage;