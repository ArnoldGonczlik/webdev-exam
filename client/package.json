{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "scripts": {
    "build": "parcel build html/index.html",
    "dev": "parcel watch html/index.html",
    "test": "jest",
    "verify": "jest --collectCoverage"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "parcel": "^2.7.0",
    "process": "^0.11.10",
    "@babel/preset-env": "^7.19.4",
    "@babel/preset-react": "^7.18.6",
    "babel-jest": "^29.2.2",
    "jest": "^29.2.2",
    "jest-environment-jsdom": "^29.2.2",
    "jsdom": "^20.0.1"
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.4.2"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "jest": {
    "testEnvironment": "jsdom",
    "globals": {
      "IS_REACT_ACT_ENVIRONMENT": true
    },
    "collectCoverageFrom": [
      "**/*.{jsx,js}",
      "!index.jsx",
      "!coverage/**",
      "!dist/**"
    ]
  }
}
