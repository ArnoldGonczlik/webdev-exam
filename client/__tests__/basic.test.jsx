import * as React from "react"
import {createRoot} from "react-dom/client";
import { act } from "react-dom/test-utils";
import { Frontpage } from "../js/frontpage.jsx";
import { MemoryRouter } from "react-router-dom";

describe("catering business tests", () => {

    it("Renders frontpage", () => {
        const element = document.createElement("div");

        const root = createRoot(element);

        act(() => {

            root.render(
                <MemoryRouter>
                    <Frontpage/>
                </MemoryRouter>
            );
        });

        expect(element.querySelector("h1")?.innerHTML).toEqual("Welcome to the fantastic catering business!");
        expect(element.innerHTML).toMatchSnapshot();

    });
});