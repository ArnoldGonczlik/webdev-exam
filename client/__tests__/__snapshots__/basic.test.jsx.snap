// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catering business tests Renders frontpage 1`] = `"<div><div><button value="/">Home</button><button value="/menu">Menu</button><button value="/login">Login</button><br><br></div><h1>Welcome to the fantastic catering business!</h1><div>We specialize in the best kind of food!</div><div>Please use the navigation at the top to look at our food selection and more</div></div>"`;
